### PR TITLE
Closes #1220: Better error/loading handling when posts can't be loaded

### LINKED ISSUE #1220: Better error/loading handling when posts can't be loaded
Right now we don't do a very good job when posts are loading, or can't be loaded form the backend:

In  src/frontend/src/components/Post/Post.jsx:

```js
  if (error) {
    console.error(`Error loading post at ${postUrl}`, error);
    return null;
  }
```

and again:

```js
  if (!post) {
    return <div>Loading...</div>;
  }
```

and in src/frontend/src/components/Posts/Posts.jsx:

```js
 // TODO: need proper error handling
  if (error) {
    console.error('Error loading post data', error);
    return null;
```

We should do a better job signaling things in the UI.

### PR DESCRIPTION
**Pull request by zjjiang2**
## Issue This PR Addresses
From Issue #1220 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [x] **UI**: Change which improves UI

## Description

Added error handling for UX, includes:
1. Post fails to load
Error
2. Post is loading
Loading
3. Timeline fails to load
PostsError

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [x] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by humphd** [comment]
Pinging some reviewers for this.  I like where it's headed, but wording might need tweaks.

**Review by cindyledev — APPROVED** [review]
This is great :D

File: src/frontend/src/components/Posts/Posts.jsx [hunk 1]
```diff
@@ -29,14 +44,29 @@ const Posts = () => {
...
+        >
+          <Grid item>
+            <SentimentDissatisfiedRoundedIcon className={classes.errorIcon} />
+          </Grid>
+          <Grid item>Blog Timeline Failed to Load!</Grid>
+        </Grid>
+      </Container>
+    );
+  } else {
+    return (
```
Comment (PedroFonsecaDEV) [inline src/frontend/src/components/Posts/Posts.jsx:64]: "Unnecessary 'else' after 'return'.eslintno-else-return"

Same as before, use "Quick fix..." and then run prettier until solve the indentation problems.

**Review by PedroFonsecaDEV — CHANGES_REQUESTED** [review]
Please don't forget to REBASE before and after start working on the requested changes. 
Thanks.

**Comment by humphd** [comment]
@zjjiang2 did you want to finish this soon?  You can probably use `npm run eslint-fix` and `npm run prettier` to sort out any issues that are left.

**Comment by PedroFonsecaDEV** [comment]
@zjjiang2 We still have some minor fixes about imports order: 

[1](https://github.com/Seneca-CDOT/telescope/pull/1251/files#diff-93a6799df46705ffa21a236b1f986e7a4d646a79b3dd6abac9103227dbf64e6dR9) and [2](https://github.com/Seneca-CDOT/telescope/pull/1251/files#diff-60a81f354235758112a505c36bc258df0296b42e471f3b13f32167a65ffb830fR9)

Then I think your PR will be good to go. 

Thanks.

File: src/frontend/src/components/Posts/Posts.jsx [hunk 2]
```diff
@@ -11,6 +12,19 @@ const useStyles = makeStyles(() => ({
     padding: 0,
     maxWidth: '785px',
   },
+  error: {
+    position: 'center',
+    color: '#B5B5B5',
```
Comment (humphd) [inline src/frontend/src/components/Posts/Posts.jsx:17]: @cindyledev, @agarcia-caicedo or someone else who knows our theme code: are there theme values we should be using for these colours instead?

File: src/frontend/src/components/Posts/Posts.jsx [hunk 3]
```diff
@@ -11,6 +12,19 @@ const useStyles = makeStyles(() => ({
...
+    fontFamily: 'Roboto',
+    fontSize: '5rem',
+    paddingBottom: '30px',
+  },
+  errorIcon: {
+    position: 'center',
+    color: '#B5B5B5',
```
Comment (humphd) [inline src/frontend/src/components/Posts/Posts.jsx:24]: Same question here re: colours/theme.

**Review by humphd — APPROVED** [review]
Looks good, with one question about theme colours vs. raw colour values.

**Review by c3ho — APPROVED** [review]

**Review by PedroFonsecaDEV — APPROVED** [review]

**Comment by PedroFonsecaDEV** [comment]
@manekenpix  
It's good to go. 
Should I click on "Merge pull request" or "Rebase and merge"?

**Comment by manekenpix** [comment]
Rebasing would be ideal so we can keep a linear history, but I don't trust github solving conflicts so I guess we'll have to merge it as is.

**Pull request merged**